### PR TITLE
Fixing handling of non-world frame velocities in setModelState.

### PR DESCRIPTION
**Pull request by ayrton04**
For issue #240

### CODE REVIEW

**Comment by hsu** [comment]
Tested with pioneer2dx model per issue #240 , looks good to me.
It'll be great if you can also add a launch file for pioneer 2dx with a test ros nodes that demonstrates the problem under a directory in `gazebo_ros/test/`? Thanks!

**Comment by ayrton04** [comment]
I'll try to throw something together soon, sure.

**Comment by hsu** [comment]
thanks @ayrton04.

**Comment by hsu** [comment]
+1

since there isn't a test yet, one more approval before I merge this.

**Comment by ayrton04** [comment]
Sorry, I've been buried lately. I'll get to it soon, I promise!

EDIT: Did you want the test in gazebo_ros/test or gazebo_plugins/test?

**Comment by hsu** [comment]
please put it in gazebo_plugins/test, thanks.

**Comment by ayrton04** [comment]
Please don't merge this or the indigo-devel PR yet. I may have discovered a different issue.

File: gazebo_ros/src/gazebo_ros_api_plugin.cpp [hunk 1]
```diff
@@ -1282,8 +1282,13 @@ bool GazeboRosApiPlugin::setModelState(gazebo_msgs::SetModelState::Request &req,
 
       //std::cout << " debug : " << relative_entity->GetName() << " : " << frame_pose << " : " << target_pose << std::endl;
       //target_pose = frame_pose + target_pose; // seems buggy, use my own
-      target_pose.pos = frame_pos + frame_rot.RotateVector(target_pos);
+      target_pose.pos = model->GetWorldPose().pos + frame_rot.RotateVector(target_pos);
```
Comment (ayrton04) [inline gazebo_ros/src/gazebo_ros_api_plugin.cpp:1285]: Part of the problem is that frame_pos is the pose of the relative_entity (e.g., the "chassis" link in the Pioneer model). This means that when we call model->SetWorldPose(target_pose) on line 1310, we'll setting the model's position using the position of the relative_entity. If relative_entity has a non-zero offset from the model origin, then we'll be repeatedly moving the model by the amount of the offset. In the case of the pioneer, the model has a Z offset of 0.16 meters, so every time this function is called, the robot climbs 16 centimeters into the air. If you call it with great enough frequency, the robot flies.

**Comment by ayrton04** [comment]
Should be fixed. I added the new test as a ROS test in the CMakelists file.

**Pull request merged**